### PR TITLE
CARTO docs update

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/6652

#### Change List
- Correct outdated documentation in CARTO module
- Update CARTO docs link in whats-new

### CODE REVIEW

File: docs/api-reference/carto/carto-layer.md [hunk 1]
```diff
@@ -153,6 +163,32 @@ Receives arguments:
...
+
+##### `MAP_TYPES.TILESET`
+
+Tiled data will be used, depending on `formatTiles`. All [`MVTLayer`](/docs/api-reference/geo-layers/mvt-layer.md) properties will be inherited.
+
+##### `MAP_TYPES.TABLE`
+
+If `format` is `FORMATS.TILEJSON`, or the table is large, dynamic tiling is enabled and the behavior is as per `MAP_TYPES.TILESET`. Otherwise behavior is as per `MAP_TYPES.QUERY`.
+
+the layer works with vector tiles if the `type` property is `MAP_TYPES.TILESET` and with GeoJSON data if the `type` is `MAP_TYPES.QUERY`. If the type is `MAP_TYPES.TABLE`.
```
Comment (borja-munoz) [inline docs/api-reference/carto/carto-layer.md:190]: Remove this line

**Review by borja-munoz — CHANGES_REQUESTED** [review]

File: docs/api-reference/carto/carto-layer.md [hunk 2]
```diff
@@ -109,6 +111,14 @@ Required when apiVersion is `API_VERSIONS.V3`.
 
 Name of the connection registered in the CARTO workspace.
 
+##### `format` (String, optional)
+
+Only supported when apiVersion is `API_VERSIONS.V3`. Use to override the default data format. Possible values are: `FORMATS.GEOJSON`, `FORMATS.JSON` and `FORMATS.TILEJSON`.
```
[the remark this answers is not in the document]
Comment (alasarr) [reply docs/api-reference/carto/carto-layer.md:116]: Don't follow you here

File: docs/api-reference/carto/carto-layer.md [hunk 3]
```diff
@@ -153,6 +163,32 @@ Receives arguments:
 
 - `error` (`Error`)
 
+### Sub Layers
```
Comment (alasarr) [inline docs/api-reference/carto/carto-layer.md:166]:
```suggestion
### SubLayers
```

[the remark this answers is not in the document]
Comment (alasarr) [reply docs/api-reference/carto/carto-layer.md]: Done

File: docs/api-reference/carto/carto-layer.md [hunk 4]
```diff
@@ -153,6 +163,32 @@ Receives arguments:
...
+
+`CartoLayer` works with the different CARTO Maps API versions (v1, v2, and v3). When using version v1 and v2, the layer always works with vector tiles so it inherits all properties from [`MVTLayer`](/docs/api-reference/geo-layers/mvt-layer.md).
+
+#### API v3
+
+When using v3, the behavior depends on the `type` property.
+
+##### `MAP_TYPES.QUERY`
+
+GeoJSON/JSON data will be used, depending on `format`. No additional properties inherited.
```
Comment (alasarr) [inline docs/api-reference/carto/carto-layer.md:180]: We should mention we're using as subLayer the GeoJSON Layer of deck.gl

File: docs/api-reference/carto/carto-layer.md (hunk 1 above)
Comment (alasarr) [reply docs/api-reference/carto/carto-layer.md:190]: I think referring to QUERY or TILESET is difficult to understand. I'd change to something like this:

Automatically based on the size of the table we decide if use tiles or a full document.

If we use a full document we use the GeoJSONLayer, if we use tiles we use a vector tile layer that extends the current MVTLayer and inherits all the properties.

You can force to use tiles by setting format to FORMATS.TILEJSON

**Review by alasarr — CHANGES_REQUESTED** [review]

File: docs/api-reference/carto/carto-layer.md [hunk 5]
```diff
@@ -153,6 +163,34 @@ Receives arguments:
...
+
+When using v3, the behavior depends on the `type` property.
+
+##### `MAP_TYPES.QUERY`
+
+GeoJSON/JSON data will be used, depending on `format`. A [`GeoJSONLayer`](/docs/api-reference/layers/geojson-layer.md) will be created.
+
+##### `MAP_TYPES.TILESET`
+
+Tiled data will be used, depending on `formatTiles`. All [`MVTLayer`](/docs/api-reference/geo-layers/mvt-layer.md) will be created and all properties will be inherited.
```
Comment (borja-munoz) [inline docs/api-reference/carto/carto-layer.md:184]: Typo here in "All MVTLayer"

**Review by alasarr — APPROVED** [review]

File: docs/api-reference/carto/carto-layer.md (hunk 1 above)
Comment (alasarr) [reply docs/api-reference/carto/carto-layer.md]: Looks great

**Pull request merged**